### PR TITLE
Add async eth.sign_transaction

### LINKED ISSUE #2827: Add async eth.sign_transaction
### What was wrong?

Some methods are still unavailable on the `AsyncEth` class.

### How can it be fixed?

Following the pattern already established, add `sign_transaction` to the `AsyncEth` class, and add corresponding tests. The `AsyncHTTPProvider` docs in `docs/provider.rst` will also need to be updated.

### PR DESCRIPTION
**Pull request by Player256**
### What was wrong?

Closes #2827

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by Player256** [comment]
@kclowes

File: web3/_utils/module_testing/eth_module.py [hunk 1]
```diff
@@ -141,6 +141,31 @@ async def test_eth_send_transaction_legacy(
         assert txn["value"] == 1
         assert txn["gas"] == 21000
         assert txn["gasPrice"] == txn_params["gasPrice"]
+    
+    @pytest.mark.asyncio
+    async def test_eth_sign_transaction(
+        self, w3: "Web3", unlocked_account: ChecksumAddress
```
Comment (kclowes) [inline web3/_utils/module_testing/eth_module.py:147]: Looks like this should be: 
```suggestion
        self, async_w3: "Web3", unlocked_account: ChecksumAddress
```

**Review by kclowes — COMMENTED** [review]
Thanks for the contribution and the ping! Looks like there are some lint errors to clean up, and I made a comment on the line where the go-ethereum tests are breaking. Let me know if you want me to make those fixes!

File: web3/_utils/module_testing/eth_module.py (hunk 1 above)
Comment (Player256) [reply web3/_utils/module_testing/eth_module.py]: I guess the problem persisted

**Comment by kclowes** [comment]
I'll take another look once we get v6 stable out. Thanks!

**Review by kclowes — APPROVED** [review]
Thanks for the PR @Player256! I fixed the failing tests and added a few more.

**Pull request merged**